nntrainer/layers/embedding.cpp

```diff
@@ -104,17 +104,19 @@ void EmbeddingLayer::forwarding(bool training) {
     float *in_data = input_.getAddress(b * input_.getDim().getFeatureLen());
 
     for (unsigned int i = 0; i < in_length; ++i) {
-      if (in_data[i] < 0)
+      if (in_data[i] > in_dim) {
+        throw std::invalid_argument("input word index is greater then in_dim");
+      }
+
+      if (in_data[i] - 1 < 0)
```

Comment:
if this is not a hard error, I guess we have to fill the output to zero?

Reply:
Does the indexing of data start from 0 or 1? If 0, then `in_data[i] == 0` is valid.

Reply:
It must be skipped.  It is just padding. No need to calculate.